### PR TITLE
 Bug 1505266 - Validate node IP is local during sdn node initialization

### PR DESCRIPTION
**Pull request by pravisankar**
Fix for https://bugzilla.redhat.com/show_bug.cgi?id=1505266

### CODE REVIEW

**Comment by pravisankar** [comment]
@openshift/networking @danwinship PTAL

File: pkg/network/node/node.go [hunk 1]
```diff
@@ -144,36 +144,17 @@ func New(c *OsdnNodeConfig) (network.NodeInterface, error) {
 		return nil, nil
 	}
 
+	if useConnTrack && c.ProxyMode != componentconfig.ProxyModeIPTables {
```
Comment (danwinship) [inline pkg/network/node/node.go]: I feel like it's better to always log the "Initializing SDN" message... can you move things around so that happens before this? (And why did you move this before `setNodeIP`? Why does that matter?)

File: pkg/network/common/common.go [hunk 2]
```diff
@@ -276,3 +278,30 @@ func RegisterSharedInformerEventHandlers(kubeInformers kinternalinformers.Shared
...
+	for _, link := range links {
+		addrs, err := netlink.AddrList(link, netlink.FAMILY_V4)
+		if err != nil {
+			glog.Warningf("Could not get addresses of interface %q: %v", link.Attrs().Name, err)
+			continue
+		}
+
+		for _, addr := range addrs {
+			if addr.IP.String() == ip {
+				return link, addr.IPNet, nil
```
Comment (danwinship) [inline pkg/network/common/common.go]: This is not the same as what findEgressLink() was doing. The `netlink.Addr`'s embedded `IPNet` is an IP address (eg "172.17.0.**3**/24"), but `eip.localEgressNet` is supposed to be the subnet CIDR that includes it (eg "172.17.0.**0**/24"). Hence the re-parsing of it (because there's no public method to directly mask the IP address form into the subnet form).

File: pkg/network/node/node.go (hunk 1 above)
Comment (pravisankar) [reply pkg/network/node/node.go]: Agree, it is always good to log this msg. 
This proxy check doesn't depend on computing hostname/selfIP, so I just moved that logic a bit up in this method.

**Comment by pravisankar** [comment]
@danwinship Updated, PTAL
